examples/forms.rs

```diff
@@ -0,0 +1,271 @@
+    let radio_group_id = Ref::new(5);
+
+    // The FormXObjects for our checkboxes need bounding boxes, in this case
+    // these are the same size as out rectangles, but within their coordinate
+    // system.
+    let bbox = Rect::new(0.0, 0.0, 30.0, 18.0);
+
+    // We define our three radio buttons, they all have a different appearance
+    // streams, but if they shared the same appearance stream and used the
+    // RADIOS_IN_UNISON flag, then two buttons could refer to the same choice.
+    // This is not widely supported, so we'll simply chowcase some normal radio
+    // buttons here.
+    //
+    // NOTE: a reader like okular will also use on-state name in the default
+    //       appearance.
+    let radios = [
+        (Ref::new(6), Rect::new(108.0, 710.0, 138.0, 728.0), b"ch1"),
+        (Ref::new(7), Rect::new(140.0, 710.0, 170.0, 728.0), b"ch2"),
+        (Ref::new(8), Rect::new(172.0, 710.0, 202.0, 728.0), b"ch3"),
+    ];
+    // First, we define the radio group parent. The children of this field will
+    // be our actual buttons. We can define most of the radio related properties
+    // here.
+    let mut field = pdf.form_field(radio_group_id);
+
+    // We set some flags to get the exact behavior we want.
+    // - FieldFlags::NO_TOGGLE_OFF means that once a button is selected it
+    //   cannot be manually turned off without turning another button on
+    // - FieldFlags::RADIOS_IN_UNISON ensures that if we have buttons which use
+    //   the same appearance on-state, they'll be toggled in unison with the
+    //   others (although we don't use this here)
+    // Finally we define the children of this field, the widget annotations
+    // which again define appearance and postion of the individual buttons.
+    //
+    // NOTE: by the time of writing this, RADIOS_IN_UNISON does not work
+    //       correctly pdf.js (firefox), okular or evince.
+    field
+        .partial_name(TextStr("radio"))
+        .field_type(FieldType::Button)
+        .field_flags(
+            FieldFlags::RADIO
+                | FieldFlags::NO_TOGGLE_TO_OFF
+                | FieldFlags::RADIOS_IN_UNISON,
+        )
+        .children(radios.map(|(id, _, _)| id));
+    field.finish();
+
+    // For buttons appearances are more relevant when printing as they're
+    // usually not as easy to find as text fields if they have no appearance.
+    let radio_on_appearance_id = Ref::new(9);
+    let radio_off_appearance_id = Ref::new(10);
+
+    // Here we prepare our appearances, the on appearance is a tick and the off
+    // appearance is empty.
+    let mut content = Content::new();
+    content.save_state();
+    content.begin_text();
+    content.set_fill_gray(0.0);
+    content.set_font(font_name2, 14.0);
+    // The character 4 is a tick in this font.
+    content.show(Str(b"4"));
+    content.end_text();
+    content.restore_state();
+    pdf.form_xobject(radio_on_appearance_id, &content.finish()).bbox(bbox);
```

Comment:
The On Appearance is not used by Quartz / Preview on Apple platforms

https://github.com/typst/pdf-writer/assets/3874949/b79ed417-d821-4ae1-8ada-a15726ff0571

**Edit:** The issue is that the symbol font was not referenced in the content stream's own resources. I'll push this change. Now, Quartz renders an off-center tick. Pdfium / Chrome will render a tick aligned with the center bottom of the box, Acrobat will only render the tick if the mouse button is held down and pdf.js still renders a bullet. The document now passes the [3-Heights PDF Validator](https://www.pdf-online.com/osa/validate.aspx) whereas before it failed.


https://github.com/typst/pdf-writer/assets/3874949/7fdb9f1b-f8fb-4044-b3c9-2784582a0756

From Quartz with the change.

https://github.com/typst/pdf-writer/assets/3874949/2e671efd-911b-4fa2-929f-2183256c7487

From Acrobat with the change.

Reply:
Interesting, thanks for testing that I'm unable to test any macOS software.